moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -521,7 +527,7 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
     }
   }
 
-  fcl::DistanceResultd fcl_result;
+  thread_local fcl::DistanceResultd fcl_result;
```

Comment:
If you make this `thread_local`, you need to call `.clear()` every time.

Reply:
I considered it but the same fields of the `DistanceResult` should be overwritten every time, so I don't actually think it's necessary.

Reply:
Hm, I'm not sure about that. It seems to me that `fcl::DistanceResult::update(...)` only overwrites the fields if the distance is lower than the previously stored one. So you'd need `clear()` to be able to get a new correct measurement.